integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/CodeValueConverter.java

```diff
@@ -55,12 +57,30 @@ public CertificateQuestionValue convert(CertificateDataElement element,
                   (CertificateDataConfigRadioMultipleCodeOptionalDropdown) element.getConfig(),
                   subQuestions
               );
+              case UE_RADIO_MULTIPLE_CODE -> convertRadioMultipleCode(
+                  value,
+                  (CertificateDataConfigRadioMultipleCode) element.getConfig()
+              );
               default -> createTextValue(value.getId());
             }
         )
         .orElse(NOT_PROVIDED_VALUE);
   }
 
+  private CertificateQuestionValueText convertRadioMultipleCode(CertificateDataValueCode value,
```

Comment:
This is duplicated from convertDropdown. Could it be converted by common functions instead?

Reply:
The list of the configs contain different objects. DropdownConfig contains DropdownItem and RadioMultipleCodeConfig contains RadioMultipleCode.

If I would break them out into common functions I would end up in a scenario where I still need to cast them both to the correct type in order to perform the ```Collectors.toMap()```.

I dont think that would make the solutions any cleaner, what do you think?

Reply:
I meant that you could try to create functions to not have to duplicate the code line by line, not that the function itself needed refactoring. I was more thinking something like this below. But if you think this makes it hard to read, it works the way you have it since it's still only a few lines which are duplicated.

```
private CertificateQuestionValueText convertRadioMultipleCode(CertificateDataValueCode value,
      CertificateDataConfigRadioMultipleCode config) {
    return convertMultipleCode(
        value,
        config.getList(),
        RadioMultipleCode::getId,
        RadioMultipleCode::getLabel
    );
  }

  private static CertificateQuestionValueText convertDropdown(
      CertificateDataValueCode value, CertificateDataConfigDropdown config) {
    return convertMultipleCode(
        value,
        config.getList(),
        DropdownItem::getId,
        DropdownItem::getLabel
    );
  }

  private static <T> CertificateQuestionValueText convertMultipleCode(
      CertificateDataValueCode value,
      List<T> list,
      Function<T, String> getId,
      Function<T, String> getLabel) {
    final var idToLabelMap = convertIdToLabelMap(list, getId, getLabel);

    return createTextValue(
        idToLabelMap.getOrDefault(value.getId(), value.getId())
    );
  }

  private static <T> Map<String, String> convertIdToLabelMap(List<T> list,
      Function<T, String> getId, Function<T, String> getLabel) {
    return list
        .stream()
        .collect(
            Collectors.toMap(
                getId,
                getLabel
            )
        );
  }
```